### ISSUE TITLE
--onsite doesn't work on dune global pool

### ISSUE THREAD

**Issue by marcmengel** [marcmengel/jobsub_lite#257]
So you need a different GLIDEIN_SITE name (FNAL_GPGRID vs Fermigrid) for dune global pool versus the jobsub pools, for obscure glidein factory  config reasons.  Should we maybe make this an environment variable setting for what site is considered "onsite", or is there /should there be a classad we can read somewhere Ii.e. on the schedd's) that will tell us what is "onsite"?   Then we wouldn't need a new code release if it changes...

**Comment by shreyb** [comment]
I actually wanted to talk about not just this issue, but also the idea of having a dune global pool "mode" (or maybe a different RPM/config install) for DUNE, so that things "just work" out of the box.  Right now, to submit to the dune global pool, they have to set `_condor_COLLECTOR_HOST`, etc.  It'd be nice if we could help them through either configuration, an easier environment variable, or a flag.

It would be easy enough to make this an environment variable, but I do like the idea of a schedd classad so we can read it dynamically.

**Comment by kherner** [comment]
In case it's useful we've been adding the GLIDEIN_DUNESite factory attribute to all sites where DUNE runs. For FermiGrid it's US_FNAL-FermiGrid. The CMS T1 is US_FNAL-T1. So you could consider looking at that when running in global pool mode.

**Comment by shreyb** [comment]
This is related to #274.  So when that closes, this will be fixed.

**Comment by shreyb** [comment]
Just talked with @njpfnal about how this would all work.  The tentative design is this.

## Setup

### Schedd

We propose that all global pool schedds that want to use jobsub_lite need to advertise the following classad attributes:
1. IsJobsubLite (already exists)
2. InDownTime (already exists)
3. Jobsub_OnsiteSiteName (NEW, name not fixed yet)
4. Jobsub_AccountingGroupTemplate (NEW, name not fixed yet)

(3) will be the site name that jobsub_lite uses if a user passes the `--onsite` flag.  It should match the `GLIDEIN_Site` classad that the glideins will advertise back to the collector.  For Fermilab, for example, this is `FNAL_GPGrid`.

(4) is a template in a yet-to-be-determined form (probably jinja, since we use it already), for the accounting group convention the global pool uses.  This is needed because, for example, GPGrid uses `group_<experiment>.<subgroup>.<user>`, whereas the DUNE Global pool uses `<subgroup>.<user>`, or (tentatively) just `<user>` if no subgroup is given.  If, for example, we do decide to have this template string be jinja2, it'd be something like `"{% if subgroup is defined and subgroup %}{{subgroup}}.{{user}}{% else %}{{user}}{% endif %}"`.

### Local 

We need to have the collector host for the global pool specified locally so jobsub_lite knows where to get the schedd list.  This can be specified in either the Condor configuration as a separate RPM with just the config file, or in our jobsub_lite configurations as environment variables.  The first is tougher to deploy, but a lot cleaner of an architecture than the second.

## Order of operations for jobsub_lite

Assume user submits something like `jobsub_submit -G dune --subgroup mars --global-pool dune file://blah`

1.  Get collector from local configuration
2. Query onsite schedd (how we determine that is TBD - maybe we need to match the IP address of a submitter's node to the schedds) for (1)-(4) above.
3. Use (3) and (4) to populate submission template

## Items to figure out before starting development in jobsub_lite code:

- [ ] How should we configure the collector name?
- [ ] How do we determine which schedd in a pool is onsite?
- [ ] Finalize names of classad attributes that schedds must advertise

**Comment by marcmengel** [comment]
So the collector name is pretty much already done -- we ask condor.  That means it can be either in the condor config files, or in `_condor_COLLECTOR_HOST ` in the environment.   We can do either way in the config rpms.

I think for determining local schedd's, we were discussing in #351 having yet another new classad on the schedd with a list of subnet masks or wildcarded domain names for hosts that can submit.    Not sure if that means we should roll #351 into this ticket or what.

**Comment by shreyb** [comment]
I think #351 should remain separate, only because it's closely related but slightly different.  That ticket is talking about saying which nodes can submit to a schedd, whereas in the design laid out above, we need to somehow specify on the schedd which domains are "onsite".  The problem lies in the fact that we might be able to submit (at some point in the future) to a schedd that's offsite, but allowed according to the standard established in #351.

I would be satisfied for the time being just matching domain-names, but that would mean that we would cut out support for, say, me installing jobsub_lite on my own laptop, having access to all the proper credentials, and then submitting to the global pool.  So I'm hoping we can come up with a different way to do it.

**Comment by shreyb** [comment]
Now, with #376 closed, this should now work.  I will test.

ETA:  Test job:  708.0@dunegpschedd01.fnal.gov 

jobsub_command:  `jobsub_submit -G dune --global-pool dune --onsite file:///usr/bin/sleep 300`

**Comment by shreyb** [comment]
We have received management direction that this item, while it should remain open, should not be a top priority.  DUNE users using the global pool are expected to operate under the assumption that everything is "offsite", and thus not use the --onsite/--offsite flags in best practice.  So while we will come back to this and fix it at some point, it does not need to be addressed immediately.
